### PR TITLE
Database test

### PR DESCRIPTION
**Pull request by Ireti203**

### CODE REVIEW

File: tests/php_hobby/test.sh [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+            echo ""
+            echo -e "1. Ensure your file contains php syntax"${NONE}
+        fi
+   done
+}
+
+check_file_content2() {    
+    for currentFile in $path_to_php_folder/*
+    do
+        if grep  "array"  $currentFile && grep "for" $currentFile
```
Comment (ToluAkin) [inline tests/php_hobby/test.sh:64]: What are you checking for here?

File: tests/php_hobby/test.sh [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+            echo -e "${BOLD}Checking the php file ...${NONE}"
+            echo -e "1. Ensure you loop through the array"${NONE}
+        fi
+   done
+}
+
+check_file_content4() {    
+    for currentFile in $path_to_php_folder/*
+    do
+        if grep "<="  $currentFile && grep "count" $currentFile
```
Comment (ToluAkin) [inline tests/php_hobby/test.sh:94]: Same as above

File: tests/php_weather/test.sh [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+path_to_php_folder="$path_to_folder/php"
+path_to_log="$submission_directory/../tests/logfile.json"
+
+check_php_folder_existence() {
+    folder_count=$(ls $path_to_folder | grep "php" -c)
+    if [ $folder_count -eq 0 ]
+    then
+        php_folder_exists=0
+        echo ""
+        echo -e "${BOLD}Checking the database folder ...${NONE}"
```
Comment (ToluAkin) [inline tests/php_weather/test.sh:38]: I think this should be a php folder check

File: tests/php_weather/test.sh [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+            echo ""
+            echo -e "1. Ensure your file contains php syntax"${NONE}
+        fi
+   done
+}
+
+check_file_content2() {    
+    for currentFile in $path_to_php_folder/*
+    do
+        if grep  "temp < 30"  $currentFile && grep "The weather is just perfect" $currentFile
```
Comment (ToluAkin) [inline tests/php_weather/test.sh:64]: Do you want to check for just `< 30` without adding `temp`?

File: tests/php_weather/test.sh [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+            echo -e "${BOLD}Checking the php file ...${NONE}"
+            echo -e "1. Ensure you use the conditions as recommended in the task criteria"${NONE}
+        fi
+   done
+}
+
+check_file_content3() {    
+    for currentFile in $path_to_php_folder/*
+    do
+        if grep  "temp >= 30"  $currentFile && grep "It's so hot today" $currentFile
```
Comment (ToluAkin) [inline tests/php_weather/test.sh:79]: Same for the rest of the checks from the above comment.

**Review by ToluAkin — CHANGES_REQUESTED** [review]
Well done @Ireti203

File: tests/php_hobby/test.sh [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+            echo -e "${BOLD}Checking the php file ...${NONE}"
+            echo -e "1. Ensure you loop through the array(s)"${NONE}
+        fi
+   done
+}
+
+check_file_content5() {    
+    for currentFile in $path_to_php_folder/*
+    do
+        if grep "["  $currentFile && grep "]" $currentFile
```
Comment (ToluAkin) [inline tests/php_hobby/test.sh:109]: You doing this check twice??

File: tests/php_hobby/test.sh [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+            echo -e "${BOLD}Checking the php file ...${NONE}"
+            echo -e "1. Ensure your code contains array(s)"${NONE}
+        fi
+   done
+}
+
+check_file_content3() {    
+    for currentFile in $path_to_php_folder/*
+    do
+        if grep  "i++"  $currentFile && grep "0" $currentFile
```
Comment (ToluAkin) [inline tests/php_hobby/test.sh:79]: What if the value they are trying to loop is not `i` or `j`?

File: tests/php_weather/test.sh [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+            echo -e "${BOLD}Checking the php file ...${NONE}"
+            echo -e "1. Ensure you use the conditions as recommended in the task criteria"${NONE}
+        fi
+   done
+}
+
+check_file_content5() {    
+    for currentFile in $path_to_php_folder/*
+    do
+        if grep "temp <= 20"  $currentFile && grep "if" $currentFile
```
Comment (ToluAkin) [inline tests/php_weather/test.sh:109]: What if the variable name is not `temp`?

**Review by ToluAkin — CHANGES_REQUESTED** [review]

File: tests/php_hobby/test.sh (hunk 6 above)
Comment (Ireti203) [reply tests/php_hobby/test.sh:109]: No, I was checking the opening and closing brackets of the array.

Comment (ToluAkin) [reply tests/php_hobby/test.sh:109]: @Ireti203 check_file_content2 and check_file_content5() are the same thing now.

**Review by ToluAkin — CHANGES_REQUESTED** [review]

File: tests/php_hobby/test.sh [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+    for currentFile in $path_to_php_folder/*
+    do
+        if grep  "{"  $currentFile && grep "}" $currentFile
+        then
+            content_exists=1
+            break     
+        else
+            content_exists=0
+            echo -e "${BOLD}Checking the php file ...${NONE}"
+            echo -e "1. Ensure you echo the appropriate result as seen in the task criteria"${NONE}
```
Comment (ToluAkin) [inline tests/php_hobby/test.sh:71]: Hi @Ireti203 This message does not correlate with the test. Kindly update for all of the checks. Thanks.

File: tests/php_hobby/test.sh [hunk 10]
```diff
@@ -0,0 +1,192 @@
+#!/bin/bash
+
+NONE='\033[00m'
+BOLD='\x1b[1m'
+ITALIC='\x1b[3m'
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+
+total_tests=5 # only 7 tests are currently being done, this value should be changed if tests increases
```
Comment (ToluAkin) [inline tests/php_hobby/test.sh:9]: I think this should be 5 tests, yeah?

**Review by ToluAkin — CHANGES_REQUESTED** [review]
This is much better. Thank you @Ireti203

**Comment by Ireti203** [comment]
I have made changes @ToluAkin

File: tests/php_hobby/test.sh [hunk 11]
```diff
@@ -0,0 +1,192 @@
+#!/bin/bash
+
+NONE='\033[00m'
+BOLD='\x1b[1m'
+ITALIC='\x1b[3m'
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+
+total_tests=5 # only 6 tests are currently being done, this value should be changed if tests increases
```
Comment (ToluAkin) [inline tests/php_hobby/test.sh:9]: @Ireti203 You didn't change the value of test in the comment

File: tests/php_hobby/test.sh [hunk 12]
```diff
@@ -0,0 +1,192 @@
...
+path_to_log="$submission_directory/../tests/logfile.json"
+
+check_php_folder_existence() {
+    folder_count=$(ls $path_to_folder | grep "php" -c)
+    if [ $folder_count -eq 0 ]
+    then
+        php_folder_exists=0
+        echo ""
+        echo -e "${BOLD}Checking the php folder ...${NONE}"
+        echo -e "${RED}Your folder should be named ${BOLD}\"php\"${NONE} ${RED}and located in the folder with your First and Last name. (${BOLD}\"$folder_name/database\"${NONE}${RED})${NONE}"
```
Comment (ToluAkin) [inline tests/php_hobby/test.sh:39]: @Ireti203 This is a php test. Why is it in a database folder?

File: tests/php_weather/test.sh [hunk 13]
```diff
@@ -0,0 +1,192 @@
+#!/bin/bash
+
+NONE='\033[00m'
+BOLD='\x1b[1m'
+ITALIC='\x1b[3m'
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+
+total_tests=6 # only 7 tests are currently being done, this value should be changed if tests increases
```
Comment (ToluAkin) [inline tests/php_weather/test.sh:9]: This too @Ireti203

**Review by ToluAkin — CHANGES_REQUESTED** [review]
Well done @Ireti203 The tests are good to go now. Thanks.

**Review by ToluAkin — APPROVED** [review]
Well done @Ireti203. Everything looks good now. Thanks.

**Pull request merged**